lib/common/output_html.c

```diff
@@ -260,6 +260,7 @@ html_output_xml(pcmk__output_t *out, const char *name, const char *buf) {
     CRM_ASSERT(out != NULL);
 
     node = pcmk__output_create_html_node(out, "pre", NULL, NULL, buf);
+    CRM_ASSERT(node != NULL);
```

Comment:
That would certainly make the patch much smaller.  I'm not really a fan of asserting in general, but I don't know that there's anything useful to do with a return value of NULL from these functions.  We might as well crash when we can control it, instead of at some unknown point later when the null pointer gets dereferenced.

Reply:
Updated.  This patch can just be dropped and the two lines needed to assert in the right place added to the type checking patch.

Reply:
Are you ready for this to be merged? I didn't see anything in the type checking patch

Reply:
The stuff in the type checking patch is just:
```
CRM_ASSERT(pcmk__str_any_of(out->fmt_name, "xml", "html", NULL));
```
in two places.  Those two functions are the basis for the whole stack of xml-related functions, so that should be sufficient.

But, I should go back and use pcmk__str_any_of in a couple more places in that patch.  I'll do that, push a new version, and then I'm okay with it being merged if you are.

Reply:
I mean in regards to the alternative to the dropped patch. But I'm fine with merging it as is. Did you mean to squash the two output helpers patches?

Reply:
Yes, I did mean to squash those two together.  Fixed and pushing again.